index.js

```diff
@@ -40,4 +37,85 @@ Babel.prototype.copyOptions = function() {
   return cloned;
 };
 
+// Promise-enabled utilities.
+
+// Recursively go through source directory and return file list.
+function deepList(dir) {
+  return new rsvp.Promise(function(resolve, reject) {
+    recreaddir(dir, function(err, files) {
+      resolve(files.map(function(file) {
+        // Normalize slashes in directories.
+        return file.replace(/\\/g, '/');
+      }));
+    });
+  });
+}
+// Read file; transpile it if it's JavaScript.
+function readAndTranspile(loc, options, ext) {
+  return new rsvp.Promise(function(resolve, reject) {
+    fs.readFile(loc, 'utf8', function(err, data) {
+      // Non-target filetype.
+      if(ext.indexOf(path.extname(loc).replace('.','')) === -1) {
+        resolve(data);
```

Comment:
so the missing piece is only  broccoli filter should likely just sync forward files it didn't match for transpile